validator/sawtooth_validator/state/client_handlers.py

```diff
@@ -1094,3 +1104,20 @@ def _respond(self, request):
         return self._wrap_response(
             endpoint=self._gossip.endpoint,
             peers=sorted(peers, key=lambda peer: peer.endpoint))
+
+
+def client_batch_submit_request_preprocessor(message_content_bytes):
+    request = client_batch_submit_pb2.ClientBatchSubmitRequest()
+
+    try:
+        request.ParseFromString(message_content_bytes)
+    except DecodeError:
+        LOGGER.error('ClientBatchSubmitRequest failed to deserialize')
+
+        return HandlerResult(
```

Comment:
Returning an undefined type in one case and a HandlerResult in another case is not a good interface to support. The return type should be consistent, which probably means creating a new PreprocessorResult that optionally wraps the processed content and signals to the Dispatcher what to do next.